src/backend/aspen/api/views/samples.py

```diff
@@ -20,12 +24,38 @@ async def list_samples(
     return False
 
 
-@router.delete("/{sample_id}")
+async def get_owned_sample_by_public_id(db, group_id, public_id, user):
```

Comment:
nit: for phylo runs, it's `get_editable_phylorun_by_id`. I think we should either standardize on `editable` or `owned` if the result is "requesting user can delete this thing". But I also might be misunderstanding the eventual permission model and we do want to clearly distinguish for other purposes later.